neo4j-plugin/src/main/java/io/cdap/plugin/neo4j/source/Neo4jDataDriveDBInputFormat.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.neo4j.source;
+
+import com.google.common.base.Throwables;
+import io.cdap.plugin.db.ConnectionConfigAccessor;
+import io.cdap.plugin.db.JDBCDriverShim;
+import io.cdap.plugin.util.DBUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.lib.db.DBConfiguration;
+import org.apache.hadoop.mapreduce.lib.db.DBInputFormat;
+import org.apache.hadoop.mapreduce.lib.db.DBWritable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Properties;
+
+/**
+ *
+ * @param <T>
+ */
+public class Neo4jDataDriveDBInputFormat<T extends DBWritable> extends DBInputFormat {
```

Comment:
why did you need to create your own inputformat?

Reply:
Existing inputformat related to work with SQL databases, however Neo4j based on CQL, that's why need create new inputformat.